content/en/apps/guides/security/securely-onboarding-users-at-scale.md

```diff
@@ -0,0 +1,140 @@
+
+No data will be lost if they log in as the _same user_.
+{{% /alert %}}
+
+### Credential list shared on internet
+
+If an online list of credentials is leaked to unauthorized parties, or works, the Internet at large in the form of being indexed by a search engine, you need to change all passwords as soon as possible for any user on the leaked list. Per the note above, change a password will log the user out immediately unless they are offline.  Having supervisors encourage a user to switch their device to offline mode (turn off all data) is a good way to ensure they can continue to deliver care so they're not locked out.
+
+### Programmatic password reset
+
+Medic [has published a script](https://github.com/medic/cht-core/blob/master/scripts/bulk-password-update-export.js) to easily change all passwords for a list of users.  Administrators will then be responsible to log CHWs back in by [securely sending](#transmitting-credentials) them their password. 
+
+Additionally, this script could be updated to immediately send a [token login](https://docs.communityhealthtoolkit.org/apps/concepts/access/#magic-links-for-logging-in-token-login) link to the user.  There would be no need to change the password as this is done automatically for you.  Note that phone numbers will need to entered for any user to receive a token login link. Here's some example `curl` calls to send a token login link for the `mary` user:
+
+```shell
+curl https://medic:password@cht.example.com/api/v1/users/mary \
+    -X POST -H "Content-Type: application/json" \
+    -d '{"token_login":true}'
+```
+
+### Audit who has changed their password
+
+If you have instructed all users to manually change their password, CHT administrators can search their server logs to see which users have made the update.  First find a list of all running services that have the word `haproxy` in it:
+
+```shell
+docker ps --filter "name=haproxy" --format '{{.Names}}'
+```
+
+Then, assuming your service is called `cht_haproxy_1`, run this command to find all password changes in the past 24 hours:
+
+```shell
+docker logs cht_haproxy_1 --since 24h 2>&1  | grep password_scheme
+```
+
+
+Here is an example event showing user `mary` having changed their password on Feb 9 18:47:
+
+`
+Feb  9 18:47:08 haproxy[12]: 172.22.0.2,couchdb,201,15,0,0,PUT,/_users/org.couchdb.user%3Amary,api,medic,'{"_id":"org.couchdb.user:mary","_rev":"3-d3dd14f3026942245ce3881adfcd513e","name":"mary","type":"user","roles":["chw"],"facility_id":"14046008-f796-4418-9ff2-12c2ef77b478","password_scheme":"***","iterations":10,"derived_key":"f846a7a20209592c613c09ea4405170735c7a557","salt":"bbb089e5fc459c2ca1088c8316e0cc99","password":"***"}',411,15,86,'node-fetch/1.0 (+https://github.com/bitinn/node-fetch)'
+`
```

Comment:
so I actually intentionally did not go for   <code>```</code> opting for <code>`</code> . This is because <code>`</code> gives us the mono-space font with different color background but wraps. the line is so comically long w/o the wrap that I figured it was worth the non-traditional use of <code>`</code>.  Further the string I'm interested in is about 1/2 down the string, make it it hard to sideways scroll to find.